src/v/storage/disk_log_impl.cc

```diff
@@ -2516,6 +2521,8 @@ disk_log_impl::remove_prefix_full_segments(truncate_prefix_config cfg) {
       },
       [this] {
           auto ptr = _segs.front();
+          // mark as tombstone before next scheduling point
+          ptr->tombstone();
           _segs.pop_front();
           _probe->add_bytes_prefix_truncated(ptr->file_size());
           return remove_segment_permanently(ptr, "remove_prefix_full_segments");
```

Comment:
There are other callers of remove_segment_permanently(), do we not need this "tombstone" in all paths that can potentially truncate an active segment.. perhaps we should make it a precondition for this method.